tests/test_timeseries.py

```diff
@@ -152,8 +152,102 @@ def test_build_from_lists(self):
         self.assertEqual(self.cts1.timezone, pytz.timezone("UTC"))
         self.assertEqual(self.cts1.name, "Test a categorical time series")
         self.assertEqual(self.cts1.type, 'CatTimeSeries')
-        
-        
+
+    def test_add_constant(self):
+        # Given
+        test_df = pd.DataFrame(columns=['ts'], index=['2020-01', '2020-02', '2020-03'])
+        test_df.loc['2020-01'] = 1.
+        test_df.loc['2020-02'] = 2.
+        test_df.loc['2020-03'] = 3.
+        ts1 = ts.TimeSeries(data=test_df, tz="Europe/London", unit='£', name="Test a time series")
+
+        expected = [4., 5., 6.]
+
+        # When
+        sum_result = ts1 + 3
+        actual = list(sum_result.data)
+
+        # Then
+        self.assertListEqual(expected, actual)
+
+    def test_add_list(self):
+        # Given
+        test_df = pd.DataFrame(columns=['ts'], index=['2020-01', '2020-02', '2020-03'])
+        test_df.loc['2020-01'] = 1.
+        test_df.loc['2020-02'] = 2.
+        test_df.loc['2020-03'] = 3.
+        ts1 = ts.TimeSeries(data=test_df, tz="Europe/London", unit='£', name="Test a time series")
+
+        expected = [4., 6., 8.]
+
+        # When
+        sum_result = ts1 + [3, 4, 5]
+        actual = list(sum_result.data)
+
+        # Then
+        self.assertListEqual(expected, actual)
+
+    def test_add_ts(self):
+        # Given
+        test_df = pd.DataFrame(columns=['ts'], index=['2020-01', '2020-02', '2020-03'])
+        test_df.loc['2020-01'] = 1.
+        test_df.loc['2020-02'] = 2.
+        test_df.loc['2020-03'] = 3.
+        ts1 = ts.TimeSeries(data=test_df, tz="Europe/London", unit='£', name="Test a time series")
+
+        # create a ts with all ones
+        ts2 = ts1 * 0 + 1
+
+        expected = [2., 3., 4.]
+
+        # When
+        sum_result = ts1 + ts2
+        actual = list(sum_result.data)
+
+        # Then
+        self.assertListEqual(expected, actual)
+
+    def test_multiply_by_constant(self):
+        # Given
+        test_df = pd.DataFrame(columns=['ts'], index=['2020-01', '2020-02', '2020-03'])
+        test_df.loc['2020-01'] = 1.
+        test_df.loc['2020-02'] = 2.
+        test_df.loc['2020-03'] = 3.
+        ts1 = ts.TimeSeries(data=test_df, tz="Europe/London", unit='£', name="Test a time series")
+
+        expected = [10., 20., 30.]
+
+        # When
+        mul_result = ts1 * 10
+        actual = list(mul_result.data)
+
+        # Then
+        self.assertListEqual(expected, actual)
+
+    def test_slicing(self):
+        # Given
+        series = pd.Series([0, 1, 2, 3, 4, 5, 6])
+        ts1 = ts.TimeSeries(data=series, tz="Europe/London", unit='£', name="Test a time series")
+
+        # slicing with start and end
+        expected = pd.Series([1, 2], index=[1, 2])
+        actual = ts1[1:3].data
```

Comment:
Would the slicing work with datetimes indices? I think it should since it is based on pandas.Series. Maybe it is worth a test.

Reply:
@FabienNugier  you can use `ts1.iloc[1:3]` for such cases (same for dataframe)